cmd/thanos/query.go

```diff
@@ -109,9 +102,6 @@ func registerQuery(app *extkingpin.App) {
 	queryTimeout := extkingpin.ModelDuration(cmd.Flag("query.timeout", "Maximum time to process query by query node.").
 		Default("2m"))
 
-	promqlEngine := cmd.Flag("query.promql-engine", "PromQL engine to use.").Default(string(promqlEnginePrometheus)).
```

Comment:
Let's not remove this flag, and modify it for getting the default engine preference of the user?